### PR TITLE
Edit other templates to extend base.html

### LINKED ISSUE #1049: Create project base template (base.html)
### Idea summary

We need a base template to use across apps. The base template should include common CSS, JS, and page elements

### Further details

- [x] create a file in `core/templates/base.html`
    - [x] add boilerplate HTML5 markup (below)
- [ ] change all other primary templates so they extend from `base.html` using `extends` and `{% block content %}` template tags (example below)
    - [ ]  https://github.com/CiviWiki/OpenCiviWiki/blob/develop/project/accounts/templates/accounts/account_base.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/develop/project/accounts/templates/accounts/register/entry_base.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/0b8e5879de1139d3b0fd77b48af0f47fb30a6051/project/accounts/templates/accounts/utils/update_settings.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/968a8df56212e88154e5129f305f798f787f4a40/project/threads/templates/threads/partials/account/login.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/968a8df56212e88154e5129f305f798f787f4a40/project/threads/templates/threads/general-message.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/968a8df56212e88154e5129f305f798f787f4a40/project/threads/templates/threads/user-setup.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/da20141a23f8f089029652f6b451d2c0c69417fe/project/threads/templates/threads/feed.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/48e8a5581f7883dd64026a9dd4cf00970d1d8434/project/threads/templates/threads/user/reset_by_email.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/d88357f4d9d71bc0ba9d0ab66e81fd469bc89740/project/threads/templates/threads/static_templates/how_it_works.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/48e8a5581f7883dd64026a9dd4cf00970d1d8434/project/threads/templates/threads/user/password_reset.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/fe74a82af787d34f3c98f085e3bf6530e46bee7b/project/threads/templates/threads/account.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/d88357f4d9d71bc0ba9d0ab66e81fd469bc89740/project/threads/templates/threads/user/settings.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/968a8df56212e88154e5129f305f798f787f4a40/project/threads/templates/threads/thread.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/d88357f4d9d71bc0ba9d0ab66e81fd469bc89740/project/threads/templates/threads/static_templates/support_us.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/da20141a23f8f089029652f6b451d2c0c69417fe/project/threads/templates/threads/static_templates/landing.html
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/blob/d88357f4d9d71bc0ba9d0ab66e81fd469bc89740/project/threads/templates/threads/static_templates/about.html
- [ ] delete the file: https://github.com/CiviWiki/OpenCiviWiki/blob/develop/project/threads/templates/threads/base/links.html
- [ ] delete all references to the `threads/base/links.html`
    - [ ] https://github.com/CiviWiki/OpenCiviWiki/search?q=%7B%25+include+%22base%2Flinks.html%22+%25%7D

### Boilerplate HTML5 markup

```html
<!doctype html>
<html>
<head>
    <meta charset="utf-8">
    <title>CiviWiki - {% block page_title %}{% endblock page_title %}</title>
    <meta name="description" content="Building a Better Democracy for the Internet Age">
    <meta name="viewport" content="width=device-width, initial-scale=1">
	
	<link href="https://fonts.googleapis.com/css?family=Lato:300,400,600,700" rel="stylesheet">

	 <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
	 <link rel="icon" type="image/png" href="/favicon-32x32.png" sizes="32x32">
	 <link rel="shortcut icon" href="/favicon.ico">
	 <meta name="msapplication-TileImage" content="/mstile-144x144.png">
	 <meta name="msapplication-TileColor" content="#fcffff"> 
	 <link type="text/css" rel="stylesheet" href="{% static "dependencies/materialize.min.css" %}">
	 <link type="text/css" rel="stylesheet" href="{% static "dependencies/magicsuggest-min.css" %}">
	 <script type="text/javascript" src="{% static "dependencies/jquery.min.js" %}"></script>
	 <script type="text/javascript" src="{% static "dependencies/underscore.js" %}"></script>
	 <script type="text/javascript" src="{% static "dependencies/backbone-min.js" %}"></script>
	 <script type="text/javascript" src="{% static "dependencies/materialize.min.js" %}"></script>
	 <script type="text/javascript" src="{% static "dependencies/magicsuggest-min.js" %}"></script>
	 <script type="text/javascript" src="{% static "js/base.js" %}"></script>
	 <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
  
    {% block extra_css %}{% endblock extra_css %}
</head>

<body>
    {% block content %}{% endblock content %}

    {% block extra_js %}{% endblock extra_js %}
</body>
</html>
```

### Extending `base.html` in a child template

In order to extend `base.html` we need to add two things in each relevant child template:

- At the top of the template, add `{% extends `base.html` %}`
- surround remaining content with `{% block content %}` and `{% endblock content %}`
- change any `<title>...</title>` tags to `{% block page_title %}...{% endblock page_title %}`
- if present, delete any `<!doctype html>`, `<head>...</head>`, or `<body>...</body>` markup
    - move the content from `<head>...</head>` to the `base.html`, unless it is already there

### PR DESCRIPTION
**Pull request by earthprelude**
- closes #1049 
- edited the 12 remaining templates listed to make them extend `core/templates/base.html`
- modified files by removing `<doctype html>`, `<html>`, `<head>` and `<body>` markup, and adding relevant template tags
- removed CiviWiki from the title of child templates when present 
- deleted the file `threads/templates/threads/base/links.html` and removed references to it 
-----------------------
- editing `entry_base.html` to extend `base.html` created 2 levels of template inheritance, because `entry_base.html` is itself the parent template of 6 other templates. To prevent the error caused by `entry_base.html` having 2 blocks with the same name (`content`), I changed the name of the empty block it defines from `content` to `entry_content`, and updated its 6 child templates to match
- removed `{% include "static_nav.html" %}` from `base.html` because several child templates include a different navigation bar (`global_nav.html`)
- returned `{% include "static_nav.html" %}` to the 4 child templates in the `core/templates` folder

### CODE REVIEW

File: project/accounts/templates/accounts/register/entry_base.html [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
                 </div>
             </div>
             <div class="back-to-main">
                 <a class="text" href="/"><i class="material-icons mini">keyboard_backspace</i>{% trans "Back to Main" %}</a>
             </div>
     </div>
 </div>
-</body>
-</html>
+{% endblock content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/register/entry_base.html:23]: We don't need to wrap all of the `div`s with `block content`. Rather, we can just have a single `block content`within the divs, as was previously the case.

```html
<div class="entry-base center-align valign-wrapper">
    <div class="entry-forms">
        <div class="card wrapper z-depth-3">
                <div class="card-content" id="entry-content">
                    {% block content %}{% endblock %}
                </div>
            </div>
            <div class="back-to-main">
                <a class="text" href="/"><i class="material-icons mini">keyboard_backspace</i>{% trans "Back to Main" %}</a>
            </div>
    </div>
</div>
```

File: project/accounts/templates/accounts/register/entry_base.html [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-    {% include "accounts/base/links.html" %}
+
+{% block extra_css %}
     {% include "accounts/base/less_headers/login_less.html" %}
-    <title>{% block title %}CiviWiki{% endblock %}</title>
-</head>
-<body>
+{% endblock extra_css %}
+
+{% block content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/register/entry_base.html:9]: Remove this block wrapper
```suggestion
```

File: project/accounts/templates/accounts/register/entry_base.html (hunk 1 above)
Comment (brylie) [inline project/accounts/templates/accounts/register/entry_base.html:23]: Remove this block wrapper
```suggestion
```

File: project/accounts/templates/accounts/register/entry_base.html [hunk 3]
```diff
@@ -1,24 +1,23 @@
...
 {% include 'accounts/register/header.html' %}
 <div class="entry-base center-align valign-wrapper">
     <div class="entry-forms">
         <div class="card wrapper z-depth-3">
                 <div class="card-content" id="entry-content">
-                    {% block content %}{% endblock %}
+                    {% block entry_content %}{% endblock %}
```
Comment (brylie) [inline project/accounts/templates/accounts/register/entry_base.html:15]: This should remain as `block content`
```suggestion
                    {% block content %}{% endblock %}
```

File: project/accounts/templates/accounts/register/register.html [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {% extends "accounts/register/entry_base.html" %}
 
-{% block title %}CiviWiki - Register{% endblock %}
+{% block page_title %}Register{% endblock %}
 
-{% block content %}
+{% block entry_content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/register/register.html:5]: This should remain as `content`
```suggestion
{% block content %}
```

File: project/accounts/templates/accounts/users/password_reset_complete.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {% extends 'accounts/register/entry_base.html' %}
 
-{% block content %}
+{% block entry_content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/users/password_reset_complete.html:3]: This should remain as `content`
```suggestion
{% block content %}
```

File: project/accounts/templates/accounts/users/password_reset_confirm.html [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {% extends 'accounts/register/entry_base.html' %}
 
-{% block content %}
+{% block entry_content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/users/password_reset_confirm.html:3]: This should remain as `content`
```suggestion
{% block content %}
```

File: project/accounts/templates/accounts/users/password_reset_done.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {% extends 'accounts/register/entry_base.html' %}
 
-{% block content %}
+{% block entry_content %}
```
Comment (brylie) [inline project/accounts/templates/accounts/users/password_reset_done.html:3]: This should remain as `content`
```suggestion
{% block content %}
```

**Review by brylie — CHANGES_REQUESTED** [review]
This looks great so far! Let's just revert the `entry_content` block to be `content`. All we have to do is remove the outer `block content` wrapper from the `entry_base.html` so there is only the remaining `block content` within the `div`s.

**Comment by earthprelude** [comment]
Hi @brylie, thanks! 

When I tried making these suggested changes, it caused the following change to the login screen: 
image
The same thing happens to the register screen:
image

It looks like the child templates of `entry_base.html` are no longer rendering properly (including `login.html` and `register.html`). The `<div>` content of `entry_base.html`, that used to be shared with all its child templates, doesn't seem to show up anymore.  

I think this problem was caused by removing the outer `block content` wrapper from `entry_base.html`. Since the `<div>` content is no longer enclosed in `{% block content %}...{% endblock %}`, it doesn't fill in the empty `{% block content %}{endblock}` of `base.html`, and doesn't get rendered on the screen. 

Since `entry_base.html` is serving 2 roles (as both child template of `base.html`, and parent template of 6 other files), I think it needs 2 separate sets of blocks to perform those 2 different roles (both child and parent). If we remove one set, the 2 levels of template inheritance don't work correctly. 

If I revert the changes and leave the files how they were (with separate `content` and `entry_content` blocks), the child templates render normally, and I can see the regular login screen again: 
image

**Comment by brylie** [comment]
OK, my bad. We'll merge your changes as they are. Thanks!

**Review by brylie — APPROVED** [review]

**Pull request merged**